webcompat/webhooks/helpers.py

```diff
@@ -17,20 +17,30 @@
 from webcompat.helpers import proxy_request
 
 
-def extract_browser_label(body):
-    """Parse the labels from the body in comment:
+def extract_metadata(body):
+    """Parse all the hidden comments for issue metadata
 
-    <!-- @browser: value -->.
-    Currently this only handles a single label,
-    because that's all that we set in webcompat.com.
+    <!-- @foo: bar -->.
+    Returns a dict with all such comments as members,
+    with foo, bar as key, value.
     """
-    match_list = re.search(r'<!--\s@(\w+):\s([^\d]+?)\s[\d\.]+\s-->', body)
-    if match_list:
-        # perhaps we do something more interesting depending on
-        # what groups(n)[0] is in the future.
-        # right now, match_list.groups(0) should look like:
-        # ('browser', 'firefox')
-        browser = match_list.groups(0)[1].lower()
+    match_list = re.findall(r'<!--\s@(\w+):\s(.+)\s-->', body)
+    # Build a dict but only for the first seen instance of a key
+    # (in case there are duplicates added by the user after)
+    metadata_dict = {}
+    for key, value in match_list:
+        if key not in metadata_dict:
+            metadata_dict[key] = value
+    return metadata_dict
+
+
+def extract_browser_label(metadata_dict):
+    """Return the browser label from metadata_dict."""
+    browser = metadata_dict.get('browser', None)
+    # Only proceed if browser looks like "FooBrowser 99.0"
+    if browser and re.search(r'([^\d]+?)\s[\d\.]+', browser):
```

Comment:
My first attempt didn't have this regexp... but that would return the wrong result for something like `Firefox Mobile` (`browser-firefox`). So.

Reply:
What about we do this instead:

```python
def extract_browser_label(metadata_dict):
    """Return the browser label from metadata_dict."""
    browser = metadata_dict.get('browser', None)
    if browser:
        browser = browser.encode('utf-8')
        browser = browser.lower()
        browser = browser.translate(None, '()0123456789.')
        browser = '-'.join(browser.split())
        browser = 'browser-{name}'.format(name=browser)
    return browser
```

And tests.

```python
>>> metadata_dict = {'browser' : 'Firefox'}
>>> extract_browser_label(metadata_dict)
'browser-firefox'
>>> metadata_dict = {'browser' : 'Firefox 30.0'}
>>> extract_browser_label(metadata_dict)
'browser-firefox'
>>> metadata_dict = {'browser' : 'Firefox Mobile 30.0'}
>>> extract_browser_label(metadata_dict)
'browser-firefox-mobile'
>>> metadata_dict = {'browser' : 'Firefox Mobile'}
>>> extract_browser_label(metadata_dict)
'browser-firefox-mobile'
>>> metadata_dict = {'browser' : 'Firefox (tablet)'}
>>> extract_browser_label(metadata_dict)
'browser-firefox-tablet'
```

Reply:
This doesn't work, `{'browser' : 'Firefox'}` should return `None`, because it's missing a version number. Same for the last 2. I think we're splitting hairs on this part, I recommend we land it as-is and if you'd like to file a follow up issue to refactor away from regexp, that's fine.

Reply:
Followup issue for this part https://github.com/webcompat/webcompat.com/issues/1949
Also we forgot to remove the  `else: return None` ;)